cors_filter.go

```diff
@@ -120,35 +133,22 @@ func (c CrossOriginResourceSharing) isOriginAllowed(origin string) bool {
 		return false
 	}
 	if len(c.AllowedDomains) == 0 {
+		if c.AllowedDomainFunc != nil {
+			return c.AllowedDomainFunc(origin)
+		}
 		return true
 	}
 
-	allowed := false
+	// exact match on each allowed domain
 	for _, domain := range c.AllowedDomains {
-		if domain == origin {
-			allowed = true
-			break
+		if domain == ".*" || domain == origin {
```

Comment:
Should `domain` and `origin` be lowercased before comparison?

Reply:
yes, i think this should happen.  RFC4343: Domain Name System (DNS) names are "case insensitive".